python/sglang/srt/models/deepseek_v2.py

```diff
@@ -1156,19 +1146,24 @@ def __init__(
             alt_stream=alt_stream,
         )
 
-        self.info = self._compute_info(config, layer_id=layer_id, is_nextn=is_nextn)
-        previous_layer_info = self._compute_info(
-            config, layer_id=layer_id - 1, is_nextn=False
+        self.is_layer_sparse = self._is_layer_sparse(layer_id, is_nextn=is_nextn)
+        is_previous_layer_sparse = self._is_layer_sparse(layer_id - 1, is_nextn=False)
+
+        self.layer_scatter_modes = LayerScatterModes.init_new(
```

Comment:
It can be handled internally in `LayerCommunicator`. With this change, `LayerScatterModes` becomes a private class of `communicator.py`.

Reply:
I make it exposed to allow in the future the LayerScatterModes will be used by modules other than LayerCommunicator, since LayerScatterModes tells us some facts about how layer input / attn / mlp / layer out shape looks like.